database/sqlserver/sqlserver.go

```diff
@@ -33,7 +33,7 @@ var (
 	ErrMultipleAuthOptionsPassed = fmt.Errorf("both password and useMsi=true were passed.")
 )
 
-var lockErrorMap = map[mssql.ReturnStatus]string{
+var lockErrorMap = map[int]string{
```

Comment:
Can you continue to use [`mssql.ReturnStatus`](https://pkg.go.dev/github.com/microsoft/go-mssqldb#readme-return-status)?

Reply:
Hi!

Sorry for the late reply!

I think `int` is the correct choice here, since we are selecting an `int` result (`DECLARE @lockResult int;`).

Note that in the previous version, this map was never used, since `status` variable was always `0` (this statement does not actually write anything to `status` variable: `ss.conn.ExecContext(context.Background(), query, aid, &status)`).

Reply:
Ahh, you're right. The [docs](https://pkg.go.dev/github.com/microsoft/go-mssqldb#ReturnStatus) mention `ReturnStatus` being used for `PROC`s